confidential-datahub/kms/src/plugins/aliyun/protobuf/dkms_api.proto

```diff
@@ -0,0 +1,236 @@
+syntax = "proto3";
+
+package dkms_api;
+
+message KmsEncryptRequest {
+  string KeyId = 1;
+  bytes Plaintext = 2;
+  bytes Aad = 3;
+}
+
+message KmsEncryptResponse {
+  string KeyId = 1;
+  bytes CiphertextBlob = 2;
+  string RequestId = 3;
+}
+
+message EncryptRequest {
+  string KeyId = 1;
+  bytes Plaintext = 2;
+  string Algorithm = 3;
+  bytes Aad = 4;
+  bytes Iv = 5;
+  string PaddingMode = 6;
+}
+
+message EncryptResponse {
+  string KeyId = 1;
+  bytes CiphertextBlob = 2;
+  bytes Iv = 3;
+  string RequestId = 4;
+  string Algorithm = 5;
+  string PaddingMode = 6;
+}
+
+message KmsDecryptRequest {
+  bytes CiphertextBlob = 1;
+  bytes Aad = 2;
+}
+
+message KmsDecryptResponse {
+  string KeyId = 1;
+  bytes Plaintext = 2;
+  string RequestId = 3;
+}
+
+message DecryptRequest {
+  bytes CiphertextBlob = 1;
+  string KeyId = 2;
+  string Algorithm = 3;
+  bytes Aad = 4;
+  bytes Iv = 5;
+  string PaddingMode = 6;
+}
+
+message DecryptResponse {
+  string KeyId = 1;
+  bytes Plaintext = 2;
+  string RequestId = 3;
+  string Algorithm = 4;
+  string PaddingMode = 5;
+}
+
+message SignRequest {
+  string KeyId = 1;
+  bytes Digest = 2;
+  string Algorithm = 3;
+  bytes Message = 4;
+  string MessageType = 5;
+}
+
+message SignResponse {
+  string KeyId = 1;
+  bytes Signature = 2;
+  string RequestId = 3;
+  string Algorithm = 4;
+  string MessageType = 5;
+}
+
+message VerifyRequest {
+  string KeyId = 1;
+  bytes Digest = 2;
+  bytes Signature = 3;
+  string Algorithm = 4;
+  bytes Message = 5;
+  string MessageType = 6;
+}
+
+message VerifyResponse {
+  string KeyId = 1;
+  bool Value = 2;
+  string RequestId = 3;
+  string Algorithm = 4;
+  string MessageType = 5;
+}
+
+message HmacRequest {
+  string KeyId = 1;
+  bytes Message = 2;
+}
+
+message HmacResponse {
+  string KeyId = 1;
+  bytes Signature = 2;
+  string RequestId = 3;
+}
+
+message GenerateRandomRequest {
+  int32 Length = 1;
+}
+
+message GenerateRandomResponse {
+  bytes Random = 1;
+  string RequestId = 2;
+}
+
+message HashRequest {
+  bytes Message = 1;
+  string Algorithm = 2;
+}
+
+message HashResponse {
+  bytes  Digest = 1;
+  string RequestId = 3;
```

Comment:
Maybe you wanted `string RequestId = 2;` here

Reply:
I'm not sure about this actually. The protos are given by the official aliyun kms sdk. Also I found some message definitions are not used at all. Let me delete them.